File: src/main/java/leets/leenk/domain/birthday/application/util/BirthdayChecker.java
package leets.leenk.domain.birthday.application.util;

import leets.leenk.domain.user.domain.entity.User;
import org.springframework.stereotype.Component;

import java.time.LocalDate;

@Component
public class BirthdayChecker {
    public boolean validateIsBirthdayToday(LocalDate birthday) {
        LocalDate today = LocalDate.now();

        return (birthday != null)
                && birthday.getMonthValue() == today.getMonthValue()
                && birthday.getDayOfMonth() == today.getDayOfMonth();
    }

    public boolean isUserBirthdayToday(User user) {
        return validateIsBirthdayToday(user.getBirthday());
    }
}

File: src/main/kotlin/leets/leenk/domain/birthday/application/util/BirthdayChecker.kt
package leets.leenk.domain.birthday.application.util

import leets.leenk.domain.user.domain.entity.User
import org.springframework.stereotype.Component
import java.time.LocalDate

@Component
class BirthdayChecker {
    fun validateIsBirthdayToday(birthday: LocalDate?): Boolean {
        val today = LocalDate.now()

        return birthday != null &&
            birthday.monthValue == today.monthValue &&
            birthday.dayOfMonth == today.dayOfMonth
    }

    fun isUserBirthdayToday(user: User): Boolean = validateIsBirthdayToday(user.birthday)
}
